query-engine/connectors/sql-query-connector/src/ordering.rs

```diff
@@ -54,33 +55,43 @@ pub fn compute_joins(
     order_by: &OrderBy,
     order_by_index: usize,
     base_model: &ModelRef,
-) -> (Vec<AliasedJoin>, Column<'static>) {
+) -> (Vec<OrderingJoin>, Column<'static>) {
     let join_prefix = format!("{}{}", ORDER_JOIN_PREFIX, order_by_index);
     let mut joins = vec![];
+    let mut order_by_column_alias: Option<String> = None;
 
-    for rf in order_by.path.iter() {
-        let (join_alias, join_data) = compute_join(base_model, rf, join_prefix.as_str());
+    if order_by.sort_aggregation.is_some() {
+        let (aggr_column_alias, ordering_join) = compute_aggr_join(order_by, join_prefix.as_str());
 
-        joins.push(AliasedJoin {
-            data: join_data,
-            alias: join_alias,
-        });
+        order_by_column_alias = Some(aggr_column_alias);
+        joins.push(ordering_join);
+    } else {
+        for rf in order_by.path.iter() {
+            let ordering_join = compute_join(base_model, rf, join_prefix.as_str());
+
+            joins.push(ordering_join);
+        }
     }
 
     // This is the final column identifier to be used for the scalar field to order by.
     // - If it's on the base model with no hops, it's for example `modelTable.field`.
     // - If it is with several hops, it's the alias used for the last join, e.g.
     //   `{join_alias}.field`
+    // - If it's with an order by aggregation, it's the alias used for the join + alias used for the aggregated column. eg:
```

Comment:
This is somewhat questionable as you actually want `count(*)` instead of counting on a field.